CorsixTH/Lua/entities/machine.lua

```diff
@@ -154,14 +175,15 @@ function Machine:calculateSmoke(room)
   local threshold = self:getRemainingUses()
 
   -- If now exploding, clear any smoke
-  if threshold < 1 then
+  if threshold < 1 and self.saved_by_extinguisher == 0 then
     setSmoke(self, false)
+    return
   -- Else if urgent repair needed
   elseif threshold < 4 then
     -- Display smoke, up to three animations per machine
     -- i.e. < 4 one plume, < 3 two plumes or < 2 three plumes of smoke
     setSmoke(self, true)
-    -- turn on additional layers of the animation for extra smoke plumes, depending on how damaged the machine is
+    -- turn on additional ldayers of the animation for extra smoke plumes, depending on how damaged the machine is
```

Comment:
oops

Reply:
Think you commented on the wrong commit here?